crates/but/src/id.rs

```diff
@@ -1,9 +1,39 @@
 use std::fmt::Display;
 
+use bstr::ByteSlice;
 use but_core::ref_metadata::StackId;
 use but_hunk_assignment::HunkAssignment;
 use gitbutler_command_context::CommandContext;
 
+pub struct IdDb {
+    unassigned: CliId,
+}
+
+impl IdDb {
+    pub fn new(ctx: &CommandContext) -> anyhow::Result<Self> {
+        let mut max_zero_count = 1; // Ensure at least two "0" in ID.
+        let stacks = crate::log::stacks(ctx)?;
```

Comment:
I see now that feature-gating the binary behind a `legacy` feature will be required to prevent usage like this to proliferate. The tell here is `gitbutler_command_context`, which is still spread throughout the codebase, and keeps spreading.

I will work on making this harder 😅, while offering alternatives.